onnxruntime/python/session.py

```diff
@@ -103,7 +132,19 @@ def run(self, output_names, input_feed, run_options=None):
             raise ValueError("Model requires {} inputs. Input Feed contains {}".format(num_required_inputs, num_inputs))
         if not output_names:
             output_names = [output.name for output in self._outputs_meta]
-        return self._sess.run(output_names, input_feed, run_options)
+        try:
+            return self._sess.run(output_names, input_feed, run_options)
+        except C.EPFail as err:
+            if self._enable_fallback:
```

Comment:
>if self._enable_fallback: [](start = 12, length = 25)

Is it possible that for CPUProvider only, if failed, it will try CPU one more time?

Reply:
no, the intent is to retry only for EPFail status, which should only get returned from EP's Compile() or EP's compute_func